### PR TITLE
Updating publish-iis to be a portable app

### PR DESCRIPTION
**Pull request by moozzyk**

### CODE REVIEW

**Comment by Tratcher** [comment]
Does this unblock it so we can actually use it now?

**Comment by moozzyk** [comment]
It does for the old publish-iis (i.e. the one that updated the config for httpPlatformHandler) if you have the right version of the runtime (as per https://github.com/dotnet/cli/issues/2071). 
It moves the new publish-iis further - I want to make other changes and tests separately.

File: samples/IISSample/project.json [hunk 1]
```diff
@@ -15,7 +14,12 @@
   "frameworks": {
     "net451": { },
     "netstandardapp1.5": {
-      "imports": "portable-net451+win8"
+      "imports": [
+        "dnxcore50", "portable-net451+win8"
+      ],
+      "dependencies": {
+        "NETStandard.Library": "1.5.0-*"
```
Comment (weshaggard) [inline samples/IISSample/project.json]: This should be `1.5.0-rc2-*`

File: src/dotnet-publish-iis/project.json [hunk 2]
```diff
@@ -14,12 +14,21 @@
     "Microsoft.Extensions.Configuration.Json": "1.0.0-*",
     "Microsoft.DotNet.ProjectModel": "1.0.0-*",
     "Microsoft.DotNet.Cli.Utils": "1.0.0-*",
-    "NETStandard.Library": "1.5.0-*",
     "System.Diagnostics.Process": "4.1.0-*"
   },
 
   "frameworks": {
-    "dnxcore50": {
+    "netstandard1.5": {
```
Comment (weshaggard) [inline src/dotnet-publish-iis/project.json]: If this is an app perhaps it should be `netcoreapp1.0`. That should also eliminate the need for the imports.

File: samples/IISSample/project.json (hunk 1 above)
Comment (moozzyk) [inline samples/IISSample/project.json]: The feed use (aspnetcidev/aspnetvolatile) only has compatible packages (i.e. we won't take rc3 packages)

File: src/dotnet-publish-iis/project.json (hunk 2 above)
Comment (moozzyk) [inline src/dotnet-publish-iis/project.json]: @weshaggard - this is a tool.

File: src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json [hunk 3]
```diff
@@ -14,16 +14,21 @@
...
+    "netstandard1.5": {
+       "imports": [
         "portable-net451+win8",
         "dnxcore50"
-      ]
+      ],
+      "dependencies": {
+        "Microsoft.NETCore.App": {
+          "type": "platform",
+          "version": "1.0.0-rc2-*"
```
Comment (moozzyk) [inline src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json]: I should actually drop rc2 from this version...

File: src/dotnet-publish-iis/project.json (hunk 2 above)
Comment (weshaggard) [inline src/dotnet-publish-iis/project.json]: It is a tool but it is referencing the Microsoft.NETCore.App meta-package trying to be a shared .NET Core app so the correct TFM is netcoreapp1.0 for those.

File: src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json (hunk 3 above)
Comment (weshaggard) [inline src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json]: You guys can decide but I would suggest keeping rc2 in the version.

File: src/dotnet-publish-iis/project.json (hunk 2 above)
Comment (moozzyk) [inline src/dotnet-publish-iis/project.json]: OK. In that case I will try changing it to `netcoreapp1.0`

Comment (weshaggard) [inline src/dotnet-publish-iis/project.json]: As an FYI this might not work until @anurse actually disabled RID inference in the cli.

Comment (troydai) [inline src/dotnet-publish-iis/project.json]: @brthor does mention in this [issue](https://github.com/dotnet/cli/issues/1970) that tools should target `netstandard1.5`. I guess the suggestion has been changed? changed?

Comment (weshaggard) [inline src/dotnet-publish-iis/project.json]: I don't believe there is anything special about tools here. Until recently portable apps needed to use netstandard1.5 but with the latest cli tools netcoreapp1.0 should be the tfm.

**Comment by moozzyk** [comment]
:up::date:
rebased on release and updated to netcoreapp1.0

File: src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json [hunk 4]
```diff
@@ -14,16 +14,21 @@
...
     "Microsoft.DotNet.Cli.Utils": "1.0.0-*",
-    "NETStandard.Library": "1.5.0-*",
     "System.Diagnostics.Process": "4.1.0-*"
   },
 
   "frameworks": {
-    "netstandardapp1.5": {
-      "imports": [
+    "netcoreapp1.0": {
+       "imports": [
```
Comment (anurse) [inline src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json]: spacing

File: src/dotnet-publish-iis/project.json [hunk 5]
```diff
@@ -14,12 +14,21 @@
...
     "Microsoft.DotNet.ProjectModel": "1.0.0-*",
     "Microsoft.DotNet.Cli.Utils": "1.0.0-*",
-    "NETStandard.Library": "1.5.0-*",
     "System.Diagnostics.Process": "4.1.0-*"
   },
 
   "frameworks": {
-    "dnxcore50": {
+    "netcoreapp1.0": {
+       "imports": [
```
Comment (anurse) [inline src/dotnet-publish-iis/project.json]: spacing

**Comment by anurse** [comment]
:shipit:

File: src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json [hunk 6]
```diff
@@ -14,16 +14,21 @@
     "Microsoft.Extensions.Configuration.Json": "1.0.0-*",
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json]: Still needed?

Comment (moozzyk) [inline src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json]: Looks like not needed anymore - removing.

File: src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json (hunk 4 above)
Comment (moozzyk) [inline src/Microsoft.AspNetCore.Server.IISIntegration.Tools/project.json]: fixed

File: src/dotnet-publish-iis/project.json (hunk 5 above)
Comment (moozzyk) [inline src/dotnet-publish-iis/project.json]: fixed

**Comment by Eilon** [comment]
cc @NTaylorMullen @davidfowl 

I thought we were going to wait to do portable apps until we had roll-forward support, no? Won't this tool just be broken anyway once we take a new CLI?

**Comment by davidfowl** [comment]
The new CLI has it and since the corefx packages haven't changed in a while this should be fine. We just can't grab a newer Microsoft.NetCore.App than the CLI that we have and @NTaylorMullen is going to gate that.

**Comment by NTaylorMullen** [comment]
@Eilon ya this went in premature. @moozzyk and I talked in person. He wasn't aware (he got it in as I was writing up an email). 

@davidfowl I beleive @Eilon is referring to the ability to have auto-roll forward support. @DamianEdwards @Eilon and I talked and the decision was to wait until we had a CLI that supported the roll forward to get portable app changes in. As it stands currently, we wont be broken unless we pull a newer shared framework; I believe that was @DamianEdwards and @Eilon's concern about getting things like this in too early (since we were blocked by the new CLI).

@ryanbrandenburg can comment on how close we are to having a CLI with the auto-rollforward capability.

**Comment by moozzyk** [comment]
So far it did not break anything. However looks like the test stopped running once test projects were converted to `netcoreapp1.0` (maybe this is why it did not break). This conversion revealed two blocking bugs - one in NuGet and one in CLI (https://github.com/dotnet/cli/issues/2347). There is a workaround for the NuGet issue but not for the CLI. @pakrym is working on both (I think the fix in NuGet has already been merged).

**Comment by davidfowl** [comment]
It won't break anything because the version of the shared framework matches the package.

**Comment by Eilon** [comment]
Yeah I'm not *super* concerned about this going in early, but I do think it'll break very soon. But hopefully we get the new CLI ASAP and all will be well.

**Comment by moozzyk** [comment]
I am ready to revert this change if it breaks. I just did not want to do this proactively because it did not break.

**Comment by Eilon** [comment]
Yeah that's fine.

**Pull request merged**